nifi-nar-bundles/nifi-elasticsearch-bundle/nifi-elasticsearch-client-service/src/test/java/org/apache/nifi/elasticsearch/integration/ElasticSearchLookupService_IT.java

```diff
@@ -0,0 +1,249 @@
+import org.apache.nifi.elasticsearch.TestSchemaRegistry;
+import org.apache.nifi.lookup.LookupFailureException;
+import org.apache.nifi.record.path.RecordPath;
+import org.apache.nifi.record.path.RecordPathResult;
+import org.apache.nifi.schema.access.SchemaAccessUtils;
+import org.apache.nifi.schemaregistry.services.SchemaRegistry;
+import org.apache.nifi.serialization.record.MapRecord;
+import org.apache.nifi.serialization.record.Record;
+import org.apache.nifi.serialization.record.RecordSchema;
+import org.apache.nifi.serialization.record.type.RecordDataType;
+import org.apache.nifi.util.TestRunner;
+import org.apache.nifi.util.TestRunners;
+import org.junit.jupiter.api.BeforeAll;
+import org.junit.jupiter.api.BeforeEach;
+import org.junit.jupiter.api.Test;
+
+import java.io.IOException;
+import java.util.ArrayList;
+import java.util.HashMap;
+import java.util.List;
+import java.util.Map;
+import java.util.Optional;
+
+import static org.junit.jupiter.api.Assertions.assertEquals;
+import static org.junit.jupiter.api.Assertions.assertNotNull;
+import static org.junit.jupiter.api.Assertions.assertTrue;
+
+public class ElasticSearchLookupService_IT extends AbstractElasticsearchIT {
+    private TestRunner runner;
+    private ElasticSearchClientService service;
+    private ElasticSearchLookupService lookupService;
+
+    @BeforeAll
+    public static void beforeAll() throws IOException {
+        setupTestData();
+    }
+
+    @BeforeEach
+    void before() throws Exception {
+        runner = TestRunners.newTestRunner(TestControllerServiceProcessor.class);
+        service = new ElasticSearchClientServiceImpl();
+        lookupService = new ElasticSearchLookupService();
+        runner.addControllerService("Client Service", service);
+        runner.addControllerService("Lookup Service", lookupService);
+        runner.setProperty(service, ElasticSearchClientService.HTTP_HOSTS, String.format("http://%s", ELASTICSEARCH_CONTAINER.getHttpHostAddress()));
+        runner.setProperty(service, ElasticSearchClientService.CONNECT_TIMEOUT, "10000");
+        runner.setProperty(service, ElasticSearchClientService.SOCKET_TIMEOUT, "60000");
+        runner.setProperty(service, ElasticSearchClientService.USERNAME, "elastic");
+        runner.setProperty(service, ElasticSearchClientService.PASSWORD, "s3cret");
+        runner.setProperty(TestControllerServiceProcessor.CLIENT_SERVICE, "Client Service");
+        runner.setProperty(TestControllerServiceProcessor.LOOKUP_SERVICE, "Lookup Service");
+        runner.setProperty(lookupService, ElasticSearchLookupService.CLIENT_SERVICE, "Client Service");
+        runner.setProperty(lookupService, ElasticSearchLookupService.INDEX, "user_details");
+        setTypeOnLookupService();
+
+        try {
+            runner.enableControllerService(service);
+            runner.enableControllerService(lookupService);
+        } catch (Exception ex) {
+            ex.printStackTrace();
```

Comment:
Do you need to print the stacktrace here, doesn't junit do that automatically for us when the Exception is thrown up the stack?